torch/onnx/utils.py

```diff
@@ -1787,64 +1758,58 @@ def _run_symbolic_function(
-            # this is inconsistent with regular op symbolic.
-            if op_name == "PythonOp":
-                inputs = (n, *inputs)
-            return symbolic_fn(g, *inputs, **attrs)
-        elif namespace == "onnx":
+        symbolic_function_group = registration.registry.get_function_group(
+            symbolic_function_name
+        )
+        if symbolic_function_group is not None:
+            symbolic_fn = symbolic_function_group.get(opset_version)
+            if symbolic_fn is not None:
+                attrs = {k: symbolic_helper._node_get(n, k) for k in n.attributeNames()}
+                if _need_symbolic_context(symbolic_fn):
+                    # TODO(justinchuby): Refactor how we check for the need of the symbolic context
+                    ctx = _exporter_states.SymbolicContext(_params_dict, env, n, block)
+                    return symbolic_fn(ctx, g, *inputs, **attrs)
+                # PythonOp symbolic need access to the node to resolve the name conflict,
+                # this is inconsistent with regular op symbolic.
+                if op_name == "PythonOp":
+                    inputs = (n, *inputs)
+                return symbolic_fn(g, *inputs, **attrs)
+
+        attrs = {
+            k + "_" + n.kindOf(k)[0]: symbolic_helper._node_get(n, k)
+            for k in n.attributeNames()
+        }
+        if namespace == "onnx":
             # Clone node to trigger ONNX shape inference
-            attrs = {
-                k + "_" + n.kindOf(k)[0]: symbolic_helper._node_get(n, k)
-                for k in n.attributeNames()
-            }
             return g.op(op_name, *inputs, **attrs, outputs=n.outputsSize())  # type: ignore[attr-defined]
-        elif _should_aten_fallback(
-            namespace, op_name, opset_version, operator_export_type
-        ):
+
+        if _should_aten_fallback(ns_op_name, opset_version, operator_export_type):
             # Direct ATen export requested
-            attrs = {
-                k + "_" + n.kindOf(k)[0]: symbolic_helper._node_get(n, k)
-                for k in n.attributeNames()
-            }
             outputs = n.outputsSize()
             attrs["outputs"] = outputs
             # `overload_name` is set for non-Caffe2 builds only
             return g.at(  # type: ignore[attr-defined]
                 op_name, *inputs, overload_name=_get_aten_op_overload_name(n), **attrs
             )
+
+        if symbolic_function_group is not None:
+            supported_version = symbolic_function_group.get_min_supported()
         else:
-            raise errors.UnsupportedOperatorError(
-                domain,
-                op_name,
-                opset_version,
-                symbolic_registry.get_op_supported_version(
-                    op_name, domain, opset_version
-                ),
-            )
+            supported_version = None
+        raise errors.UnsupportedOperatorError(
```

Comment:
Is this indented to the right position?

Reply:
I think it is intended

Reply:
Does this raise work with if/else above it? How does this work?

Reply:
The old else branch concludes the if branching logic in the old code. We first removed the else and dedented the code. Then the new if decides if the symbolic name is supported in other versions. Finally the error is raised. In other words, the old else and the new if does different things.

Reply:
Ahh, I see. If you enter the if/else above, the error will be raised for sure. the logic was only deciding `supported_version `

Reply:
Do you think it would be better if we add comment to separate this if/else pair with others? Cause the first glance I thought this if/else is just as important as other ifs. (But I also have a bad reputation on comment things so... just for suggestion.)

Reply:
I updated to 

```python
            symbolic_function_group.get_min_supported()
            if symbolic_function_group
```

which I think is clearer and it rids the if block.

Reply:
Just some thoughts, it might worth for revisit to abstract out the effect of operator export type.

Reply:
Create https://github.com/pytorch/pytorch/issues/85170